text_generation/causal_lm/cpp/continuous_batching/library/src/model_runner.hpp

```diff
@@ -21,234 +21,114 @@ class ModelRunner {
             std::vector<Sequence::CPtr> running_sequences = sequence_group->get_running_sequences();
+            size_t num_running_sequences = running_sequences.size();
             size_t num_scheduled_tokens = sequence_group->get_num_scheduled_tokens();
-            size_t group_position_id = sequence_group->get_num_processed_tokens(), group_context_len = group_position_id + 1;
-
-            // report to plugin a length of current continuous sub-sequence
-            subsequence_lens_data[i] = num_scheduled_tokens;
+            size_t num_blocks = sequence_group->get_num_blocks();
+            size_t group_position_id = sequence_group->get_num_processed_tokens(),
+                // spec: In case of multiple input tokens for current sequence (prompt_len > 1), context_len corresponds to first token within subgroup of scheduled tokens
+                group_context_len = group_position_id + 1;
 
-            for (size_t seq_id = 0; seq_id < running_sequences.size(); ++seq_id) {
+            for (size_t seq_id = 0; seq_id < num_running_sequences; ++seq_id) {
                 Sequence::CPtr sequence = running_sequences[seq_id];
-                size_t position_id = group_position_id, context_len = group_context_len;
-
-                for (size_t token_id = 0; token_id < num_scheduled_tokens; ++token_id, ++position_id, ++context_len) {
-                    const std::vector<KVCacheBlock::Ptr> & kv_blocks = scheduler_output.m_block_tables.at(sequence->get_id());
-                    const size_t num_blocks = kv_blocks.size();
-                    for (size_t block_id = 0; block_id < kv_blocks.size(); ++block_id)
-                        block_tables_data[block_id] = kv_blocks[block_id]->get_index();
-                    block_tables_data += max_num_blocks;
-
-                    position_ids_data[token_id] = position_id;
-                    context_lens_data[token_id] = context_len;
 
+                for (size_t token_id = 0, position_id = group_position_id; token_id < num_scheduled_tokens; ++token_id, ++position_id) {
                     // compute token for current sequence
                     input_ids_data[token_id] = position_id < sequence_group->get_prompt_len() ?
                         sequence_group->get_prompt_ids()[position_id] :
                         sequence->get_generated_ids()[position_id - sequence_group->get_prompt_len()];
 
-                    // compute slot_id
-                    size_t physical_block_id = position_id / m_scheduler_config.block_size, block_offset = position_id % m_scheduler_config.block_size;
-                    int64_t slot_id = m_scheduler_config.block_size * kv_blocks[physical_block_id]->get_index() + block_offset;
-                    slot_mapping_data[token_id] = slot_id;
+                    position_ids_data[token_id] = position_id;
                 }
 
-                // apply strides to shift to next sequence
+                context_lens_data[0] = group_context_len;
```

Comment:
I suppose here should be `context_lens_data[0] = group_context_len - num_scheduled_tokens + 1;` according to the spec: In case of multiple input tokens for current sequence (prompt_len > 1), context_len corresponds to first token within subgroup of scheduled tokens

Reply:
I've updated it at the initialization:
```
            size_t group_position_id = sequence_group->get_num_processed_tokens(),
                // spec: In case of multiple input tokens for current sequence (prompt_len > 1), context_len corresponds to first token within subgroup of scheduled tokens
                group_context_len = group_position_id + 1;
```

Reply:
thank you for review and help with fix!